pkg/storage/iterator.go

```diff
@@ -385,6 +385,7 @@ func fetchLazyChunks(ctx context.Context, chunks []*chunkenc.LazyChunk) error {
 	}
 	level.Debug(log).Log("msg", "loading lazy chunks", "chunks", totalChunks)
 
+	invalidChunks := []*chunk.Fetcher{}
```

Comment:
Reason for creating this array is later when we have to remove invalid chunks, we can iterate only invalid chunks rather than iterating all fetchers and chunks

Reply:
No need for this.